src/commands/Invoke.ts

```diff
@@ -1,27 +1,37 @@
-import fetch, { Request } from "node-fetch";
+import fetch, { BodyInit, Request } from "node-fetch";
+import { ReadStream } from "fs";
+import queryString from 'query-string';
+import { flatten } from 'q-flat';
 
 import validateAndRefresh from "./RefreshIfNeeded";
 import retrieveImpersonatedCredentials from "./ImpersonateCredentials";
 
 import { Credentials, CredentialStorage } from "../oauth";
 import { ApiConfiguration } from "../ApiConfiguration";
-import queryString from 'query-string';
-import { flatten } from 'q-flat';
+
+export type InvokeBody = Record<string, unknown> | [] | ReadStream | Buffer;
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
-const createRequest = (url: string, config: ApiConfiguration, credentials: Credentials, payload?: Record<string, unknown> | []): Request => {
+const createRequest = (url: string, config: ApiConfiguration, credentials: Credentials, payload?: InvokeBody): Request => {
     const headers = {
         "Accept": "application/json",
         "Content-Type": "application/x-www-form-urlencoded",
         "Authorization": `Bearer ${credentials.accessToken}`
     };
+    let body: BodyInit | undefined;
+    if (payload && (payload instanceof ReadStream || payload instanceof Buffer)) {
```

Comment:
Again referring to trade-chat/image/upload, payload may be an object, whose one key "file" is binary, but "trade_hash" is a plain string. I guess here we need to recursively iterate payload's value to determine for sure what mime and how payload should be handled.

Reply:
No need to iterate to determinate the mimes, just need to use `FormData`.